homeassistant/components/intent.py

```diff
@@ -0,0 +1,178 @@
+"""Module to coordinate user intentions."""
+import asyncio
+import logging
+
+import voluptuous as vol
+
+from homeassistant.core import callback
+from homeassistant.exceptions import HomeAssistantError
+
+
+DOMAIN = 'intent'
+_LOGGER = logging.getLogger(__name__)
+
+SLOT_SCHEMA = vol.Schema({
+    vol.Optional('type'): str,
+})
+
+SPEECH_TYPE_PLAIN = 'plain'
+SPEECH_TYPE_SSML = 'ssml'
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Setup intent."""
+    hass.intent = IntentRegistry(hass)
```

Comment:
Alright, it now follows the dispatcher pattern and has become a helper. No more component, no more setting of things on hass object and no more dependencies 👍